lib/util/cache/repository/impl/base.ts

```diff
@@ -1,16 +1,74 @@
-import type { RepoCache, RepoCacheData } from '../types';
+import { promisify } from 'util';
+import zlib from 'zlib';
+import hasha from 'hasha';
+import { GlobalConfig } from '../../../../config/global';
+import { logger } from '../../../../logger';
+import {
+  CACHE_REVISION,
+  isValidRev10,
+  isValidRev11,
+  isValidRev12,
+} from '../common';
+import type { RepoCache, RepoCacheData, RepoCacheRecord } from '../types';
 
-export class RepoCacheBase implements RepoCache {
-  protected data: RepoCacheData = {};
+const compress = promisify(zlib.brotliCompress);
+const decompress = promisify(zlib.brotliDecompress);
+
+export abstract class RepoCacheBase implements RepoCache {
+  protected platform = GlobalConfig.get('platform')!;
+  private oldHash: string | null = null;
+  private data: RepoCacheData = {};
+
+  protected constructor(protected readonly repository: string) {}
+
+  protected abstract read(): Promise<string | undefined>;
+
+  protected abstract write(data: RepoCacheRecord): Promise<void>;
 
-  // istanbul ignore next
   async load(): Promise<void> {
```

Comment:
I think my naming wasn't good, maybe it's a good chance to rename methods to something like `load/save` and `encode/decode` (or `compress/decompress`)

Reply:
Or maybe, `loadRaw/saveRaw` for the abstract methods above

Reply:
I do like the `save/load` for cache methods, and `read/write` for the actual reading/writing to fs/storage.

Reply:
me too

Reply:
Well, okay